### PR TITLE
[SPARK-18923][DOC][BUILD] Support skipping R/Python API docs

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

We can build Python API docs by `cd ./python/docs && make html for Python` and R API docs by `cd ./R && sh create-docs.sh for R` separately. However, `jekyll` fails in some environments.

This PR aims to support `SKIP_PYTHONDOC` and `SKIP_RDOC` for documentation build in `docs` folder. Currently, we can use `SKIP_SCALADOC` or `SKIP_API`. The reason providing additional options is that the Spark documentation build uses a number of tools to build HTML docs and API docs in Scala, Python and R. Specifically, for Python and R,

- Python API docs requires `sphinx`.
- R API docs requires `R` installation and `knitr` (and more others libraries).

In other words, we cannot generate Python API docs without R installation. Also, we cannot generate R API docs without Python `sphinx` installation. If Spark provides `SKIP_PYTHONDOC` and `SKIP_RDOC` like `SKIP_SCALADOC`, it would be more convenient.

## How was this patch tested?

Manual.

**Skipping Scala/Java/Python API Doc Build**
```bash
$ cd docs
$ SKIP_SCALADOC=1 SKIP_PYTHONDOC=1 jekyll build
$ ls api
DESCRIPTION R
```

**Skipping Scala/Java/R API Doc Build**
```bash
$ cd docs
$ SKIP_SCALADOC=1 SKIP_RDOC=1 jekyll build
$ ls api
python
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@dongjoon-hyun Just to make sure, could we actually build each individually, `cd ./python/docs && make html` for Python and `cd ./R && sh create-docs.sh` for R? (FWIW, I support this PR even if so because now these can be done in a single place).

BTW, I believe it would be nicer if the options are documented in https://github.com/apache/spark/blob/master/docs/README.md

**Comment by dongjoon-hyun** [comment]
Ur, that's true. The effective benefit might be tiny. I'll keep a few days and close this if there is no response from the others.
Thank you for review.

**Comment by srowen** [comment]
I think it's fine to make this change for consistency and convenience. It's minor. It'd be nice to document them in the README.md, briefly.

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen ! I will update the `README.md` , too.

**Comment by holdenk** [comment]
I think for new Python or Scala developers there is a benefit in skipping the R docs because they require some extra setup work to build and having the process simplified would be good for newcomers as in many projects documentation changes are how new contributors start.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @holdenk !

**Review by srowen — APPROVED** [review]
Looks good. This also changes the working directory in these parts of the scripts but I agree it's more natural to work relative to `docs/`. the changes look correct.

**Comment by srowen** [comment]
Merged to master

**Comment by dongjoon-hyun** [comment]
Thank you for merging, @srowen .
